src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -186,7 +189,7 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
                         this.gradingScaleExists = true;
                         this.gradingScale = gradingScaleResponse.body!;
                         this.isBonus = this.gradingScale!.gradeType === GradeType.BONUS;
-                        this.hasBonus = !!this.studentResults?.some((studentResult) => studentResult?.gradeWithBonus);
+                        this.hasBonus = this.studentResults?.find((studentResult) => studentResult?.gradeWithBonus)?.gradeWithBonus?.bonusStrategy;
```

Comment:
what is the type of `bonusStrategy`. We used to cast this into a boolean with `!!`. Do you think this might be beneficial to keep here?

Reply:
The type was changed from a boolean to a BonusStrategy (e.g. POINTS or GRADES_CONTINUOUS).

It can still be used as a boolean in other places that also used this value before (undefined || null === false and true otherwise) but it's also possible to distinguish different bonus types now. On the exam's score page, you can see that we can now display "Bonus Points" instead of "Bonus Grade" if applicable :)